File: spring-cloud-integrated-demo/account-service/src/test/java/com/github/jccode/springcloud/integrateddemo/account/service/AccountServiceTest.java
package com.github.jccode.springcloud.integrateddemo.account.service;

import com.github.jccode.springcloud.integrateddemo.account.model.Account;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.mybatis.spring.boot.test.autoconfigure.MybatisTest;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.context.annotation.Import;
import org.springframework.test.context.ActiveProfiles;
import org.springframework.test.context.junit4.SpringRunner;


import static org.hamcrest.MatcherAssert.assertThat;
import static org.hamcrest.Matchers.greaterThan;
import static org.hamcrest.Matchers.is;
import static org.hamcrest.Matchers.nullValue;

@SpringBootTest
@RunWith(SpringRunner.class)
@MybatisTest
@Import(AccountService.class)
@ActiveProfiles("test")
public class AccountServiceTest {

    @Autowired
    private AccountService accountService;

    @Test
    public void mybatisWorks() {
        Account account = new Account();
        account.setName("Hello");
        account.setUserId(1);
        account.setBalance(10);
        assertThat(account.getId(), is(nullValue()));

        int save = accountService.save(account);

        assertThat(save, greaterThan(0));
        assertThat(account.getId(), greaterThan(0));
    }
}

File: spring-cloud-integrated-demo/account-service/account-service-app/src/test/scala/com/github/jccode/springcloud/integrateddemo/account/service/AccountServiceTest.scala
package com.github.jccode.springcloud.integrateddemo.account.service

import com.github.jccode.springcloud.integrateddemo.account.model.Account
import org.hamcrest.MatcherAssert.assertThat
import org.hamcrest.Matchers.{greaterThan, is, nullValue}
import org.junit.Test
import org.junit.runner.RunWith
import org.mybatis.spring.boot.test.autoconfigure.MybatisTest
import org.scalatest.Matchers
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.context.annotation.Import
import org.springframework.test.context.ActiveProfiles
import org.springframework.test.context.junit4.SpringRunner

@SpringBootTest
@RunWith(classOf[SpringRunner])
@MybatisTest
@Import(Array(classOf[AccountService]))
@ActiveProfiles(Array("test"))
class AccountServiceTest extends Matchers {

  @Autowired
  private val accountService: AccountService = null

  @Test
  def mybatisWorks(): Unit = {
    val account = new Account
    account.setName("Hello")
    account.setUserId(1)
    account.setBalance(10)
    account.getId should be (null)
    val ret = accountService.save(account)
    ret should be > 0
    account.getId().toInt should be > 0
  }
}
